dataland-frontend/src/components/forms/CreateSfdrDataset.vue

```diff
@@ -0,0 +1,308 @@
+import ProductionSitesFormField from "@/components/forms/parts/fields/ProductionSitesFormField.vue";
+import { objectDropNull } from "@/utils/UpdateObjectUtils";
+import { smoothScroll } from "@/utils/SmoothScroll";
+import { DocumentToUpload, uploadFiles } from "@/utils/FileUploadUtils";
+import MostImportantProductsFormField from "@/components/forms/parts/fields/MostImportantProductsFormField.vue";
+import { Subcategory } from "@/utils/GenericFrameworkTypes";
+import ProcurementCategoriesFormField from "@/components/forms/parts/fields/ProcurementCategoriesFormField.vue";
+
+export default defineComponent({
+  setup() {
+    return {
+      getKeycloakPromise: inject<() => Promise<Keycloak>>("getKeycloakPromise"),
+    };
+  },
+  name: "CreateSfdrDataset",
+  components: {
+    SubmitButton,
+    SubmitSideBar,
+    UploadFormHeader,
+    SuccessMessage,
+    FailMessage,
+    FormKit,
+    Card,
+    PrimeButton,
+    Calendar,
+    YesNoFormField,
+    InputTextFormField,
+    FreeTextFormField,
+    NumberFormField,
+    DataPointFormField,
+    DateFormField,
+    SingleSelectFormField,
+    MultiSelectFormField,
+    NaceCodeFormField,
+    AddressFormField,
+    RadioButtonsFormField,
+    YesNoNaFormField,
+    PercentageFormField,
+    ProductionSitesFormField,
+    MostImportantProductsFormField,
+    ProcurementCategoriesFormField,
+    UploadReports,
+  },
+  directives: {
+    tooltip: Tooltip,
+  },
+  emits: ["datasetCreated"],
+  data() {
+    return {
+      formId: "createSFDRForm",
+      waitingForData: true,
+      dataDate: undefined as Date | undefined,
+      companyAssociatedSfdrData: {} as CompanyAssociatedDataSfdrData,
+      sfdrDataModel: sfdrDataModel,
+      route: useRoute(),
+      message: "",
+      smoothScroll: smoothScroll,
+      uploadSucceded: false,
+      postSfdrDataProcessed: false,
+      messageCounter: 0,
+      checkCustomInputs,
+      documents: new Map() as Map<string, DocumentToUpload>,
+      referencedReportsForPrefill: {} as { [key: string]: CompanyReport } | undefined,
+      namesOfAllCompanyReportsForTheDataset: [] as string[],
+    };
+  },
+  computed: {
+    yearOfDataDate: {
+      get(): string {
+        const currentDate = this.companyAssociatedSfdrData.data?.general?.general?.fiscalYearEnd;
+        if (currentDate === undefined) {
+          return "";
+        } else {
+          const currentDateSegments = currentDate.split("-");
+          return currentDateSegments[0] ?? new Date().getFullYear();
+        }
+      },
+      set() {
+        // IGNORED
+      },
+    },
+    subcategoryVisibility(): Map<Subcategory, boolean> {
+      const map = new Map<Subcategory, boolean>();
+      for (const category of this.sfdrDataModel) {
+        for (const subcategory of category.subcategories) {
+          map.set(
+            subcategory,
+            subcategory.fields.some((field) => field.showIf(this.companyAssociatedSfdrData.data)),
+          );
+        }
+      }
```

Comment:
We have a util method "createSubcategoryVisibilityMap()" that you can use here.

Reply:
changed